core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java

```diff
@@ -158,6 +168,11 @@ public void onSubchannelState(ConnectivityStateInfo state) {
     for (Subchannel removedSubchannel : removedSubchannels) {
       shutdownSubchannel(removedSubchannel);
     }
+
+    // Update reused subchannels with possibly updated EquivalentAddressGroups
+    for (EquivalentAddressGroup addressGroup : unchangedAddrs) {
+      subchannels.get(addressGroup).updateAddresses(resolvedAddresses.getAddresses());
```

Comment:
No, no. This updates each subchannel to point to all addresses. It should be closer to:
```java
subchannels.get(addressGroup).updateAddresses(addressGroup);
```

Reply:
Done